Sample/EventsVersioning/EventsVersioning.Tests/ExplicitSerialization/ShoppingCart.cs

```diff
@@ -0,0 +1,413 @@
+using System.Text.Json;
+using System.Text.Json.Nodes;
+
+namespace EventsVersioning.Tests.ExplicitSerialization;
+
+using static ShoppingCartEvent;
+
+public abstract class StronglyTypedValue<T>: IEquatable<StronglyTypedValue<T>> where T : notnull
```

Comment:
interesting related things https://github.com/jet/equinox/blob/ebca2357c12f5ef46bbbba6e417d153272798285/samples/Store/Domain/Infrastructure.fs#L8

(I tend not to get into using these - I just use FSharp.UMX instead of all this...

Reply:
I'll just reference it to https://event-driven.io/en/using_strongly_typed_ids_with_marten/. It's more about showing how this can be done, but I also wouldn't suggest doing that. In general, I think that strongly typed ids is quite often overkill :P